test/server/cards/02_SHD/units/ZoriiBlissValiantSmuggler.spec.ts

```diff
@@ -0,0 +1,51 @@
+describe('Zorii Bliss', function() {
+    integration(function(contextRef) {
+        describe('Zorii Bliss\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['zorii-bliss#valiant-smuggler'],
+                        hand: ['volunteer-soldier'],
+                        deck: ['battlefield-marine', 'wampa', 'pyke-sentinel']
+                    },
+                    player2: {
+                        hand: ['vanquish']
+                    }
+                });
+            });
+
+            it('draws a card on attack and discards a card at the start of the regroup phase even if zorii dies', function () {
```

Comment:
Can we add one more case where the player has no cards to discard at the end of the phase and confirm that nothing breaks